File: backend/schemas/pathology_templates.py
from pydantic import BaseModel
from typing import Optional


class PathologyTemplateSummary(BaseModel):
    id: int
    nombre: str
    categoria: str
    kcal_objetivo: float
    descripcion: Optional[str] = None
    tiempos_por_dia: int
    imagen_url: Optional[str] = None
    total_recetas: int

    class Config:
        from_attributes = True


class PathologyTemplateDetail(BaseModel):
    id: int
    nombre: str
    categoria: str
    kcal_objetivo: float
    descripcion: Optional[str] = None
    tiempos_por_dia: int
    imagen_url: Optional[str] = None
    weekly_menu: dict

    class Config:
        from_attributes = True


class AssignTemplateRequest(BaseModel):
    patient_id: int

File: frontend/src/api/pathologyTemplates.ts
import { api } from './client'

export interface PathologyTemplateSummary {
  id: number
  nombre: string
  categoria: string
  kcal_objetivo: number
  descripcion: string | null
  tiempos_por_dia: number
  imagen_url: string | null
  total_recetas: number
}

export interface PathologyTemplateDetail extends Omit<PathologyTemplateSummary, 'total_recetas'> {
  weekly_menu: { semana: unknown[] }
}

export async function listPathologyTemplates(params?: { categoria?: string; kcal_min?: number; kcal_max?: number }) {
  const { data } = await api.get<PathologyTemplateSummary[]>('/pathology-templates', { params })
  return data
}

export async function getPathologyTemplate(templateId: number) {
  const { data } = await api.get<PathologyTemplateDetail>(`/pathology-templates/${templateId}`)
  return data
}

export async function assignPathologyTemplate(templateId: number, patientId: number) {
  const { data } = await api.post<{ plan_id: number }>(`/pathology-templates/${templateId}/assign`, { patient_id: patientId })
  return data
}
